src/Applications/RTFRevit/RevitTestClient.cs

```diff
@@ -0,0 +1,166 @@
+﻿using System;
+using System.Net;
+using System.Net.Sockets;
+using System.Collections.Generic;
+using System.Diagnostics;
+using Autodesk.Revit.Attributes;
+using Autodesk.Revit.DB;
+using Autodesk.Revit.UI;
+using RTF.Framework;
+
+namespace RTF.Applications
+{
+    [Transaction(TransactionMode.Manual)]
+    [Regeneration(RegenerationOption.Manual)]
+    [Journaling(JournalingMode.UsingCommandData)]
+    public class RTFClientStartCmd : IExternalCommand
+    {
+        private static int iPort = 0;
+        private static Socket clientSocket;
+        private static IPAddress ipAddress;
+        private static IPEndPoint endPoint;
+
+        /// <summary>
+        /// The entry point when the command runs. This command will try to connect to localhost
+        /// at a port passed in through the journaling data.
+        /// </summary>
+        /// <param name="cmdData"></param>
+        /// <param name="message"></param>
+        /// <param name="elements"></param>
+        /// <returns></returns>
+        public Result Execute(ExternalCommandData cmdData, ref string message, ElementSet elements)
+        {
+            if (!ReadDataFromJournal(cmdData.JournalData))
+            {
+                message = "The port is not given or is in a bad format!";
+                return Result.Failed;
+            }
+
+            ipAddress = IPAddress.Parse(CommonData.LocalIPAddress);
+            endPoint = new IPEndPoint(ipAddress, iPort);
+            clientSocket = new Socket(AddressFamily.InterNetwork, SocketType.Stream, ProtocolType.Tcp);
+
+            clientSocket.Connect(endPoint);
+            NotifyClientStarted();
+
+            return Result.Succeeded;
+        }
+
+        /// <summary>
+        /// Read the port data from the journal data
+        /// </summary>
+        /// <param name="dataMap"></param>
+        /// <returns></returns>
+        private bool ReadDataFromJournal(IDictionary<string, string> dataMap)
+        {
+            if (!dataMap.ContainsKey("Port"))
```

Comment:
You can use dataMap.TryGetValue() which returns bool if key is present and the value as out param if present. It will skip the lookup twice.